sdcm/nemesis.py

```diff
@@ -496,28 +496,51 @@ def disrupt_nodetool_refresh(self, big_sstable=True, skip_download=False):
-                            hash_expected=sstable_md5, retries=2,
-                            user_agent=creds['user_agent'])
+            keys_num = 501000
 
         self.log.debug('Prepare keyspace1.standard1 if it does not exist')
-        self._prepare_test_table(ks='keyspace1')
+        self._prepare_test_table(ks='keyspace1', table='standard1')
         result = self.target_node.run_nodetool(sub_cmd="cfstats", args="keyspace1.standard1")
-        if result is not None and result.exit_status == 0:
-            result = self.target_node.remoter.run("sudo ls -t /var/lib/scylla/data/keyspace1/")
+
+        def do_refresh(node):
+            if not skip_download:
+                key_store = KeyStore()
+                creds = key_store.get_scylladb_upload_credentials()
+                # Download the sstable files from S3
+                remote_get_file(node.remoter, sstable_url, sstable_file,
+                                hash_expected=sstable_md5, retries=2,
+                                user_agent=creds['user_agent'])
+            result = node.remoter.run("sudo ls -t /var/lib/scylla/data/keyspace1/")
             upload_dir = result.stdout.split()[0]
-            self.target_node.remoter.run('sudo tar xvfz {} -C /var/lib/scylla/data/keyspace1/{}/upload/'.format(
+            node.remoter.run('sudo tar xvfz {} -C /var/lib/scylla/data/keyspace1/{}/upload/'.format(
                 sstable_file, upload_dir))
-            self.target_node.run_nodetool(sub_cmd="refresh", args="-- keyspace1 standard1")
-            cmd = "select * from keyspace1.standard1 where key=0x32373131364f334f3830"
-            result = self.target_node.run_cqlsh(cmd)
+            self.log.debug(f'Loading {keys_num} keys to {node.name} by refresh')
+            node.run_nodetool(sub_cmd="refresh", args="-- keyspace1 standard1")
+
+        if result is not None and result.exit_status == 0:
+            query_cmd = "SELECT * FROM keyspace1.standard1 LIMIT 1"
+            result = self.target_node.run_cqlsh(query_cmd)
+            # Add columns if they don't exist, otherwise the sstable files won't loaded by cluster by refresh
+            for col in ['C1', 'C2', 'C3', 'C4']:
+                if f'| {col} ' not in result.stdout:
+                    alter_cmd = f'ALTER TABLE keyspace1.standard1 ADD \\"{col}\\" blob'
+                    self.target_node.run_cqlsh(alter_cmd)
+            # Drop one special key before refresh, we will verify refresh by query in the end
+            delete_cmd = "DELETE FROM keyspace1.standard1 WHERE key=0x32373131364f334f3830"
+            self.target_node.run_cqlsh(delete_cmd)
+            query_verify = "SELECT * FROM keyspace1.standard1 WHERE key=0x32373131364f334f3830"
+            result = self.target_node.run_cqlsh(query_verify)
+            assert '(0 rows)' in result.stdout, 'Expect the key does not exist before `nodetool refresh`'
+
+            for node in self.cluster.nodes:
```

Comment:
If We want to use repair and only load the sstable files to on db node.

The RF should be set to be same as #cluster_nodes_number, otherwise only part of data will be loaded to the cluster (not all)

The keys aren't in the token range will be ignored, the repair won't distribute the other keys to other nodes.

Reply:
Are you sure?

Reply:
> Are you sure?

Yes.  I verified by manual testing. and It's same as Shlomi said.